java/cdk-environment/src/main/java/sleeper/environment/cdk/networking/NetworkingStack.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.environment.cdk.networking;
+
+import software.amazon.awscdk.Stack;
+import software.amazon.awscdk.StackProps;
+import software.amazon.awscdk.services.ec2.*;
+import software.constructs.Construct;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+public class NetworkingStack extends Stack {
+
+    private final Vpc vpc;
+
+    public NetworkingStack(Construct scope, StackProps props) {
+        super(scope, props.getStackName(), props);
+
+        vpc = Vpc.Builder.create(this, "Vpc")
```

Comment:
The "Vpc" string is used in other classes - could be put into a constant?

Reply:
Can you think of a reason to keep them the same?

Reply:
Just to reduce the number of magic strings dotted around

Reply:
They're setting IDs for CDK to use to refer to the resources. The different usages are for different resources, so it makes sense to set it specifically for each case.

Annoyingly, for VpcParameter I can imagine you might want two different values if you're looking up multiple VPCs, although the CDK really isn't clear about what that ID is for when you're looking up an existing VPC. I guess I'll move the lookup into the stack definition class to make sure it's not a problem.

Reply:
I've moved the lookup.